src/lib/Parser.hs

```diff
@@ -816,3 +872,6 @@ eolf = void eol <|> eof
 failIf :: Bool -> String -> Parser ()
 failIf True s = fail s
 failIf False _ = return ()
+
+_debug :: Show a => String -> Parser a -> Parser a
+_debug s m = mapReaderT (Text.Megaparsec.Debug.dbg s) m
```

Comment:
what is this for?

Reply:
I kept finding myself debugging weird megaparsec error messages, so I lifted out the debug utility from megaparsec to help with them. You can wrap a parser with this (i.e. `_debug "foo" foo` instead of just `foo`) and it prints out a bunch of diagnostic info with `Debug.Trace` whenever that wrapped parser succeeds or fails. I found it helpful, so I thought maybe it was worth merging in in case it's needed later.

Reply:
Nice, thanks for explaining that